python/ray/tune/ray_trial_executor.py

```diff
@@ -31,21 +32,32 @@ def __init__(self, queue_trials=False):
         self._avail_resources = Resources(cpu=0, gpu=0)
         self._committed_resources = Resources(cpu=0, gpu=0)
         self._resources_initialized = False
+        self._cached_runner = None
         if ray.is_initialized():
             self._update_avail_resources()
 
     def _setup_runner(self, trial):
-        cls = ray.remote(
-            num_cpus=trial.resources.cpu,
-            num_gpus=trial.resources.gpu,
-            resources=trial.resources.custom_resources)(
-                trial._get_trainable_cls())
+        if self._cached_runner is not None:
+            logger.debug("Reusing cached runner {}".format(
+                self._cached_runner))
+            existing_runner = self._cached_runner
+            self._cached_runner = None
```

Comment:
if we're processing a new trial and the trial resources are different, we can't just used the _cached_runner right?

Reply:
That's right. I think we can probably assume they are the same though, if the reuse_actors flag is manually activated.